### PR TITLE
fix: changed `mod`, add `rem` (L-03)

### LINKED ISSUE #271: [L-03]: Signed mod Function Name Does Not Reflect Truncated Remainder Semantics

### PR DESCRIPTION
**Pull request by immrsd**
#### Summary                                                                         
                                                                                              
- Renamed `SD29x9::mod` to `SD29x9::rem` to clearly communicate its truncated remainder       
semantics (sign follows the dividend).
- Redefined `SD29x9::mod` to use Euclidean remainder semantics, where the result is always  
non-negative.

#### PR Checklist

- [x] Tests
- [x] Documentation
- [x] Changelog

## Summary by CodeRabbit

* **New Features**
  * Added `rem` function for truncated remainder operations where the sign follows the dividend.

* **Breaking Changes**
  * `mod` function now uses Euclidean remainder semantics, guaranteeing non-negative results. Previous truncated remainder behavior is available via the new `rem` function.

* **Tests**
  * Updated test coverage for new and modified remainder operations across various sign and edge case scenarios.

### CODE REVIEW

**Review by ericnordelo — APPROVED** [review]
LGTM!

File: math/fixed_point/sources/sd29x9/sd29x9_base.move [hunk 1]
```diff
@@ -270,13 +270,40 @@ public fun lte(x: SD29x9, y: SD29x9): bool {
...
+/// #### Returns
+/// - The Euclidean remainder of `x` divided by `y`, always non-negative.
+/// - Returns `0` when `x` is an exact multiple of `y`.
+///
+/// #### Aborts
+/// - Aborts if `y` is zero.
+public fun mod(x: SD29x9, y: SD29x9): SD29x9 {
+    let x = decompose(x.unwrap());
+    let y = decompose(y.unwrap());
+    let remainder = x.mag % y.mag;
```
Comment (0xNeshi) [inline math/fixed_point/sources/sd29x9/sd29x9_base.move:298]: Reminder to add explicit y-equals-zero check when [PR #289](https://github.com/OpenZeppelin/contracts-sui/pull/289/changes#diff-95359e7450872a0bf549130c17a487e208e005465a962f94548425d74b6f7c36R280) is merged

**Review by 0xNeshi — APPROVED** [review]

**Review by bidzyyys — APPROVED** [review]
Looks good!

**Review by 0xNeshi — APPROVED** [review]

**Review by ericnordelo — APPROVED** [review]
